circuit-codegen/src/main/kotlin/com/slack/circuit/codegen/CircuitSymbolProcessorProvider.kt

```diff
@@ -101,16 +114,22 @@ private fun Resolver.loadOptionalKSType(name: String?): KSType? {
 private class CircuitSymbolProcessor(
   private val logger: KSPLogger,
   private val codeGenerator: CodeGenerator,
+  private val options: Map<String, String>,
 ) : SymbolProcessor {
 
   override fun process(resolver: Resolver): List<KSAnnotated> {
     val symbols = CircuitSymbols.create(resolver) ?: return emptyList()
+    val codegenMode =
+      CodegenMode.entries.find { it.name.equals(options[CIRCUIT_CODEGEN_MODE], ignoreCase = true) }
+        ?: CodegenMode.ANVIL
```

Comment:
If they've defined a mode but it's unrecognized, let's error instead of silently falling back to Anvil